ibis/expr/operations.py

```diff
@@ -696,6 +708,14 @@ class Mean(Reduction):
     output_type = rules.scalar_output(_mean_output_type)
 
 
+class Quantile(Reduction):
+
+    input_type = [value,
+                  number(name='quantile', allow_boolean=False),
+                  string(name='interpolation', optional=True)]
+    output_type = rules.scalar_output(_array_reduced_type)
```

Comment:
This is correct for multiple quantiles, but not for a single quantile. In this case the `output_type` is actually the same as the *second* argument: `T -> T` or `array<T> -> array<T>`. `rules.type_of_arg(1)` should cover this case.